### PR TITLE
Adding get project by project key

### PR DESCRIPTION
**Pull request by ld-kyee**
I'm unclear why there were so many formatting changes in the api.yaml file.

### CODE REVIEW

File: internal/dev_server/api/server.gen.go
Comment (mike-zorn) [inline internal/dev_server/api/server.gen.go:3]: hmmmm how did you install the code generator? It looks like somehow you're using an old version.

File: internal/dev_server/dev_server.go
Comment (mike-zorn) [inline internal/dev_server/dev_server.go:45]: Good thinking!

**Review by mike-zorn — COMMENTED** [review]
api yaml format changes are fine. Let's get the tool version figured out before you merge though.

File: internal/dev_server/api/server.gen.go
Comment (ld-kyee) [reply internal/dev_server/api/server.gen.go:3]: 🤦 I thought I just did a `go install` but I had so many weird things going on locally.  Let me fix this.

Comment (mike-zorn) [reply internal/dev_server/api/server.gen.go:3]: Oh the docs are kinda not great because they first tell you how to install the deprecated version. Try `go install github.com/oapi-codegen/oapi-codegen/v2/cmd/oapi-codegen@latest`

**Review by mike-zorn — APPROVED** [review]
approving, assuming we get the tool version straight.

**Pull request merged**
